WordPress/gutenberg
# Add Block Pattern Categories support

closes #22141 
closes #20816 

Adds support for Block Pattern Categories:

- Pattern Categories are registered using a separate API from block categories.
- A Block Pattern can have multiple categories.

**Todo**

 - [x] Performance when switching to the patterns tab.

**Testing instructions**

 - You can open the inserter
 - Switch  to the patterns tab
 - Make sure the patterns are organized per category
 - Make sure you can still search and insert patterns.
